### PR TITLE
gradle.yml: Upgrade Codecov Action

### LINKED ISSUE #127: Upgrade CodeCov usage
See https://github.com/reposense/RepoSense/issues/1734
Seems to apply to AB3 as well

### PR DESCRIPTION
**Pull request by yhtMinceraft1010X**
Fixes #127 

Proposed Commit Message:
```
Codecov GitHub Action v1 uses a deprecated bash uploader, which may
cease to function soon. The v2 version also has a files arg instead of a
file arg, suggesting possible deprecation of the file arg.

Let's upgrade to v2, use the files arg for coverage.xml and the directory
arg for the directory in which coverage.xml is found.

```

### CODE REVIEW

**Comment by yhtMinceraft1010X** [comment]
I tried running CanIHasReview but kept getting the same error for each attempt. Here is an issue describing the error for one such attempt: https://github.com/CanIHasReview/CanIHasReview/issues/37

**Comment by damithc** [comment]
> I tried running CanIHasReview but kept getting the same error for each attempt. Here is an issue describing the error for one such attempt: [CanIHasReview/CanIHasReview#37](https://github.com/CanIHasReview/CanIHasReview/issues/37)

@yhtMinceraft1010X while we wait for a response from CanIHasReview maintainer, we can process this PR without CanIHasReview.

@se-edu/tech-team-level1 for your review please ...

**Review by LimJunxue — APPROVED** [review]
LGTM

**Review by Py0000 — APPROVED** [review]
LGTM! 👍

**Review by JoelHo — APPROVED** [review]
LGTM!

File: .github/workflows/gradle.yml
Comment (chownces) [inline .github/workflows/gradle.yml]: Did a quick scan through the [codecov README](https://github.com/codecov/codecov-action) and realized that the `file` argument might have been deprecated.

This argument was removed from the README over a year ago in this [commit](https://github.com/codecov/codecov-action/commit/25b873867a4d9c029691df2b437b0c7c7ba410cd).

Didn't have time to check if `file` is still working correctly. Should we consider shifting over to `files` instead?

Comment (yhtMinceraft1010X) [reply .github/workflows/gradle.yml]: I tried out the `files` arg. It seems to work.

**Review by JinHao-L — APPROVED** [review]
LGTM :+1:

**Comment by damithc** [comment]
@yhtMinceraft1010X Are the three commits independent incremental changes? If not, they should be in the same commit.

**Comment by damithc** [comment]
Thanks for the PR @yhtMinceraft1010X 
Confirmed `file`/`files`/`directory` are called 'flags'? 
Can revise the 2nd paragraph of the proposed commit message to remove the reference to RepoSense (as that project is not related to AB3)?

**Comment by yhtMinceraft1010X** [comment]
> Thanks for the PR @yhtMinceraft1010X Confirmed `file`/`files`/`directory` are called 'flags'? Can revise the 2nd paragraph of the proposed commit message to remove the reference to RepoSense (as that project is not related to AB3)?

I realised the proper term should be `args`. I've made the changes in the proposed commit message.

**Pull request merged**
